test/e2e/remediation_test.go

```diff
@@ -99,85 +94,64 @@ func remediation() {
-		if len(provisioningBMHs) != 2 {
-			return errors.New("Waiting for 2 BMHs to be Provisioned")
-		}
-		return nil
+		g.Expect(provisioningBMHs).To(HaveLen(2))
 	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
 
 	By("Scaling up machine deployment to 3 replicas")
 	scaleMachineDeployment(ctx, bootstrapClient, clusterName, namespace, 3)
 
 	By("Waiting for one BMH to start provisioning")
-	Eventually(func(g Gomega) error {
+	Eventually(func(g Gomega) {
 		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
-		if err != nil {
-			Logf("Error: %v", err)
-			return err
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 		g.Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioning)).To(HaveLen(1))
-		return nil
-	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...)
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
 
 	Logf("Verifying that only one BMH starts provisioning")
-	Consistently(func() error {
+	Consistently(func(g Gomega) {
 		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
-		if err != nil {
-			Logf("Error: %v", err)
-			return err
-		}
-		Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioned)).To(HaveLen(3))
-		Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioning)).To(HaveLen(1))
-		return nil
-	}, e2eConfig.GetIntervals(specName, "monitor-provisioning")...)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioned)).To(HaveLen(3))
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioning)).To(HaveLen(1))
+	}, e2eConfig.GetIntervals(specName, "monitor-provisioning")...).Should(Succeed())
 
 	Logf("Annotating BMH as healthy")
 	annotateBmh(ctx, bootstrapClient, workerBmh, unhealthyAnnotation, nil)
 
 	Byf("Waiting for all (%d) BMHs to be Provisioned", allMachinesCount)
-	Eventually(func(g Gomega) error {
+	Eventually(func(g Gomega) {
 		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
-		if err != nil {
-			Logf("Error: %v", err)
-			return err
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 		g.Expect(filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateProvisioned)).To(HaveLen(allMachinesCount))
-		return nil
-	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...)
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
 
 	By("Waiting for all Machines to be Running")
-	Eventually(func(g Gomega) error {
+	Eventually(func(g Gomega) {
 		machines := clusterv1.MachineList{}
 		g.Expect(bootstrapClient.List(ctx, &machines, client.InNamespace(namespace))).To(Succeed())
 		g.Expect(filterMachinesByPhase(machines.Items, "Running")).To(HaveLen(allMachinesCount))
-		return nil
-	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...)
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
```

Comment:
I think this is where the flake was coming from. We did not have `Should(Succeed())` and just returned `nil`.